cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -138,6 +150,22 @@ private void generateShardConfig() {
       if (externalView == null) {
         monitor.incrementConfigGeneratorNullExternalView();
         LOG.error("Failed to get externalView for resource: " + resource);
+        /**
+         * In some situations, we may encounter a null externalView for a given resource.
+         * This can happen, since there exists a race condition between retrieving list of resources
+         * and then iterating over each of those resources to retrieve corresponding externalView
+         * and potential deletion of a resource in between.
+         *
+         * Another situation where we may receive null externalView for a resource is in case where
+         * a resource is newly created but it's externalView has not yet been generated by helix
+         * controller.
+         *
+         * There can be quite a few race conditions which are difficult to enumerate and under such
+         * scenarios, we can't guarantee that externalView for a given resource exists at a specific
+         * moment. In such cases, it is safe to ignore such resource until it's externalView is
+         * accessible and not null.
```

Comment:
would it be possible that we get a null external view for an existing active resource? In that case, should we use the old shard map for this resource?

Reply:
That shouldn't be a case.

Reply:
good to know that.